# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../xorg-server/bf333c2f9833a178887e7bdd7fc338f1e09c387f.patch
# Copyright (C) 2009 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

From bf333c2f9833a178887e7bdd7fc338f1e09c387f Mon Sep 17 00:00:00 2001
From: John Tapsell <john.tapsell@john-desktop.(none)>
Date: Wed, 03 Sep 2008 11:20:17 +0000
Subject: fbdevhw: Remove pixclock check.

The check can fail because the output from FBIOGET_VSCREENINFO is used to set
Clock in fbdev2xfree_timing().  Then in fbdevHWSetMode(), xfree2fbdev_timing()
is called which sets the pixclock based on Clock.  The resulting circle results
in slight rounding errors, causing the comparision check in fbdev_modes_equal
to fail.
---
diff --git a/hw/xfree86/fbdevhw/fbdevhw.c b/hw/xfree86/fbdevhw/fbdevhw.c
index 4d51422..5269277 100644
--- a/hw/xfree86/fbdevhw/fbdevhw.c
+++ b/hw/xfree86/fbdevhw/fbdevhw.c
@@ -245,7 +245,6 @@ fbdev_modes_equal(struct fb_var_screeninfo *set, struct fb_var_screeninfo *req)
 		set->green.length == req->green.length &&
 		set->blue.length == req->blue.length &&
 		set->xres == req->xres && set->yres == req->yres &&
-		set->pixclock == req->pixclock &&
 		set->right_margin == req->right_margin &&
 		set->hsync_len == req->hsync_len &&
 		set->left_margin == req->left_margin &&
--
cgit v0.8.2
